fog/ingest/server/src/controller.rs

```diff
@@ -1027,20 +1042,103 @@ where
         &self,
         ingress_public_key: &CompressedRistrettoPublic,
         state: &mut MutexGuard<IngestControllerState>,
-    ) -> Result<IngressPublicKeyStatus, Error> {
+    ) -> Result<Option<IngressPublicKeyStatus>, Error> {
+        let next_block_index = state.get_next_block_index();
+        let current_pubkey_expiry_window = state.get_pubkey_expiry_window();
+        let current_pubkey_expiry = next_block_index + current_pubkey_expiry_window;
+
+        let previous_block_index = next_block_index - 1;
+        let previous_pubkey_expiry_window = state.get_previous_pubkey_expiry_window();
+        let previous_pubkey_expiry = previous_block_index + previous_pubkey_expiry_window;
+
+        let is_active = state.is_active();
+
+        // TODO: Remove this log line on final code push
+        log::info!(
+            self.logger,
+            "===> {} (next_block_index {} + pubkey_window {} = current {}) vs. (prev_block_index {} + prev_window {} = previous {}) {} {}",
+            is_active,
+            next_block_index,
+            current_pubkey_expiry_window,
+            current_pubkey_expiry,
+            previous_block_index,
+            previous_pubkey_expiry_window,
+            previous_pubkey_expiry,
+            state.get_last_report_published_block_index(),
+            state.get_next_report_published_block_index()
+        );
+
+        // In order to publish a report, the previous block index's pubkey expiry
+        // must be smaller than the current pubkey expiry.
+        // This if-logic checks for this.
+        // ex.
+        // If block_index=6 with window=100 gives a pubkey_expiry=106
+        //
+        // On the next block index if we reduce the window, so:
+        //    block_index=7 with window=10 gives a pubkey_expiry=17
+        // ...we do not publish, b/c the previous index's pubkey_expiry (106)
+        //    is bigger than the current index's pubkey_expir (17).
+        //
+        // However, if we would have increased the window, so:
+        //    block_index=7 with window=102 gives a pubkey_expiry=109
+        // ...we do publish, b/c the previous index's pubkey_expiry (106)
+        //    is smaller than the current index's pubkey_expiry (109).
+        //
+        // We don't want to publish on every index, only every N indexes,
+        // where N is the size of the window.
+        // ex.
+        // If block_index=1 with window=100, we publish a report on indexes
+        // 101, 201, 301, 401, etc.
```

Comment:
The previous logic just checked if the previous pubkey_expiry was smaller than the current pubkey_expiry, but we still don't want to publish a report every single time.  So I amended the if-logic below as an attempt to only publish every N indexes (where N is the size of the window).

Reply:
@garbageslam is that true? Why would we not want to publish a new report every index?
Publishing every window size seems like a meaningful change to me so I am wondering what is the reason behind it.
What happens if we fail to publish at this reduced interval? Seems like we'll need a much more aggressive retry policy. We could also run into an issue where the consensus network is slightly ahead of fog, in which case users attempting to pay fog recipients will have to wait until fog ingest catches up (e.g. if our window is 100 and we're starting at block 50, we will publish at 50, 150, 250, etc. At block 149 a user might grab a report that is only valid for one block and consensus might already be past it, while ingest might still be at for example block 140 chugging along towards 150 but until it gets there users cannot make payments).

Reply:
To add to that, my personal preference is to keep the changeset to a minimum. If we do want to reduce the publishing frequency we can tackle that in a followup PR.

Reply:
oh hmm..I thought this ticket was to do just that (reduce publishing frequency).  But only in the case that we increase the window.  Right now, the controller state's set_pubkey_expiry_window() function doesn't let you reduce the window.  So I thought that was the separate ticket if we wanted to handle that case.  Let me see if we can get some clarification.